src/do_command_helpers.py

```diff
@@ -0,0 +1,245 @@
+        "cmd": YDN_DECONFIGURE,
+        "sender_id": service_id.hex,
+    }
+    return command_dict
+
+
+def _normalize_bbox(
+    width: int, height: int, xmin: float, ymin: float, xmax: float, ymax: float
+) -> Tuple[int, int, int, int]:
+    """
+    Convert normalized bbox coordinates (0-1 floats) to pixel values.
+
+    Args:
+        width (int): Frame width
+        height (int): Frame height
+        xmin (float): Normalized x min (0-1)
+        ymin (float): Normalized y min (0-1)
+        xmax (float): Normalized x max (0-1)
+        ymax (float): Normalized y max (0-1)
+
+    Returns:
+        Tuple[int, int, int, int]: bbox as (xmin, ymin, xmax, ymax) in pixels
+    """
+    normVals = np.array([width, height, width, height])
+    bbox = np.array([xmin, ymin, xmax, ymax])
+    normalized_bbox = (np.clip(bbox, 0, 1) * normVals).astype(int)
+    return tuple(normalized_bbox)
+
+
+def encode_detections(
+    dets: dai.ImgDetections, labels: List[str], sensor: Sensor
+) -> List[Dict]:
+    """
+    Encodes dai.ImgDetections obj to a List[Dict] that can be sent
+    via do_command.
+
+    Args:
+        dets (dai.ImgDetections): the DepthAI detections obj
+        labels (List[str]): list of labels with indices corresponding to label num from ImgDetection
+        sensor (Sensor): sensor config
+
+    Returns:
+        List[Dict]: List of dictionaries containing detection data
+    """
+    encoded_detections = []
+    for det in dets.detections:
+        xmin, ymin, xmax, ymax = _normalize_bbox(
+            sensor.width, sensor.height, det.xmin, det.ymin, det.xmax, det.ymax
+        )
+        dictionary = {
+            "x_min": xmin,
+            "y_min": ymin,
+            "x_max": xmax,
+            "y_max": ymax,
+            "confidence": det.confidence,
+            "class_name": labels[det.label],
+        }
+        encoded_detections.append(dictionary)
+    return encoded_detections
+
+
+def decode_detections(det_list: List) -> List[Detection]:
+    """
+    Convert a list of detection dicts from do_command to Detection objects.
+
+    Args:
+        det_list (List): List of detection dicts
+
+    Returns:
+        List[Detection]: List of Detection objects
+    """
+    viam_dets: List[Detection] = []
+    for det_dict in det_list:
+        viam_det = Detection(
+            x_min=int(det_dict["x_min"]),
+            x_max=int(det_dict["x_max"]),
+            y_min=int(det_dict["y_min"]),
+            y_max=int(det_dict["y_max"]),
+            confidence=det_dict["confidence"],
+            class_name=det_dict["class_name"],
+        )
+        viam_dets.append(viam_det)
+    return viam_dets
+
+
+def encode_image_data(img_bytes: bytes) -> Dict:
+    """
+    Encode image bytes to a base64 dict compatible with do_command.
+
+    Args:
+        img_bytes (bytes): Image bytes
+
+    Returns:
+        Dict: Encoded image data
+    """
+    img_data_dict = {
+        "bytes": base64.b64encode(img_bytes).decode("ascii"),
```

Comment:
[q] Do we need to encode image as a string or can we send over raw binary with do_command?

Reply:
I tried raw binary. Unfortunately does not recognize binary as an accepted `ValueType`

Reply:
ok, was just checking because I am pretty sure I was able to send binary over do_command with gosdk.

Reply:
`8/13/2024, 1:40:05 PM error rdk zap/options.go:212 finished unary call with code Unknown error rpc error: code = Unknown desc = ValueError - Unexpected type - file_name='/Users/seanyu/Projects/viam-camera-oak/.venv/lib/python3.12/site-packages/google/protobuf/internal/well_known_types.py' func_name='_SetStructValue' line_num=471 grpc.code Unknown`

Reply:
weird. Prob a Python SDK thing

Reply:
Let's keep a mental note that we do this and if we need to optimize performance based on customer request we can revisit?

Reply:
Yeah sounds good.